ios/Classes/SwiftCustomerIoPlugin.swift

```diff
@@ -7,59 +7,63 @@ import CioMessagingInApp
 public class SwiftCustomerIoPlugin: NSObject, FlutterPlugin {
     
     private var methodChannel: FlutterMethodChannel!
+    private var inAppMessagingChannelHandler: CusomterIOInAppMessaging!
     
     public static func register(with registrar: FlutterPluginRegistrar) {
         let instance = SwiftCustomerIoPlugin()
         instance.methodChannel = FlutterMethodChannel(name: "customer_io", binaryMessenger: registrar.messenger())
         registrar.addMethodCallDelegate(instance, channel: instance.methodChannel)
+        
+        instance.inAppMessagingChannelHandler = CusomterIOInAppMessaging(with: registrar)
     }
     
     deinit {
         self.methodChannel.setMethodCallHandler(nil)
+        self.inAppMessagingChannelHandler.detachFromEngine()
     }
     
     public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
         switch(call.method) {
```

Comment:
I don't have anything in mind to suggest on this right now but this looks very tough to read on Github at least 😅 May be some formatting might help cleaning this up a little? 🤓

Reply:
Yes, that looks better 😅 Although I think it would be even better if it was formatted like follows:

```swift
    call.toNativeMethodCall(result: result) {
        initialize(params: $0)
    }
```

But if the current formatting is how swift formatting works then I'm fine with it and we can simply resolve this conversation 😄